core/src/executor/evaluate/mod.rs

```diff
@@ -292,6 +292,10 @@ async fn evaluate_inner<'a, 'b: 'a, 'c: 'a, T: GStore>(
             let indexes = try_join_all(indexes.iter().map(eval)).await?;
             expr::array_index(obj, indexes)
         }
+        Expr::Array { elem } => {
+            let elem = try_join_all(elem.iter().map(eval)).await?;
+            expr::array(elem)
+        }
```

Comment:
now it doesn't need to split the evaluation code in `evaluate/expr.rs`.
then it would be simply enough with
```rust
Expr::Array { elem } => {
            try_join_all(elem.iter().map(eval))
                .await?
                .into_iter()
                .map(Value::try_from)
                .collect::<Result<Vec<_>>>()
                .map(Value::List)
                .map(Evaluated::Value)
        }
```

Reply:
Thx, i will apply the following